pkgs/tools/system/s0ix-selftest-tool/default.nix

```diff
@@ -0,0 +1,80 @@
+{
+  acpica-tools,
+  bash,
+  bc,
+  coreutils,
+  fetchFromGitHub,
+  gawk,
+  gnugrep,
+  gnused,
+  linuxPackages,
+  lib,
+  pciutils,
+  powertop,
+  resholve,
+  stdenv,
+  util-linux,
+  xorg,
+  xxd,
+}:
+resholve.mkDerivation {
+  pname = "s0ix-selftest-tool";
+  version = "unstable-2022-11-04";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "S0ixSelftestTool";
+    rev = "1b6db3c3470a3a74b052cb728a544199661d18ec";
+    hash = "sha256-w97jfdppW8kC8K8XvBntmkfntIctXDQCWmvug+H1hKA=";
+  };
+
+  # don't use the bundled turbostat binary
+  patchPhase = ''
+    rm turbostat
+    substituteInPlace s0ix-selftest-tool.sh --replace '"$DIR"/turbostat' 'turbostat'
+  '';
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    chmod +x s0ix-selftest-tool.sh
+    cp s0ix-selftest-tool.sh $out/bin/s0ix-selftest-tool
+  '';
+
+  solutions = {
+    default = {
+      scripts = ["bin/s0ix-selftest-tool"];
+      interpreter = "${lib.getExe bash}";
+      inputs = [
+        acpica-tools
+        bc
+        coreutils
+        gawk
+        gnugrep
+        gnused
+        linuxPackages.turbostat
```

Comment:
Or shall this package also be under `linuxPackages`, in order to use the correct version of turbostat for all kernel version.

Reply:
I considered following that pattern, and can still move it there if necessary. Upstream is bundling their own turbostat which is being used across all kernel versions, which implies to me the specific kernel version doesn't matter.

Reply:
There appears to be little (not tied to kernel version) references to linuxPackages in nixpkgs. But I do find one in `pkgs/desktops/gnome/misc/gnome-applets/default.nix`, I suppose this is acceptable as long as it works.

Reply:
Yeah, not much prior art on this. It does build and work as-is in my testing, so I would vote for leaving it standalone rather than putting under every kernel version.